src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/diagnostics/rules/NoLocalesInJavadocUrlDiagnosticRule.java

```diff
@@ -0,0 +1,57 @@
+package com.azure.tools.apiview.processor.diagnostics.rules;
+
+import com.azure.tools.apiview.processor.diagnostics.DiagnosticRule;
+import com.azure.tools.apiview.processor.model.APIListing;
+import com.azure.tools.apiview.processor.model.Diagnostic;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.nodeTypes.NodeWithJavadoc;
+
+import java.util.Arrays;
+import java.util.Locale;
+
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.getClasses;
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.getPublicOrProtectedConstructors;
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.getPublicOrProtectedFields;
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.getPublicOrProtectedMethods;
+import static com.azure.tools.apiview.processor.analysers.util.ASTUtils.makeId;
+import static com.azure.tools.apiview.processor.model.DiagnosticKind.WARNING;
+
+public class NoLocalesInJavadocUrlDiagnosticRule implements DiagnosticRule {
+
+    private final String[] languageTags;
+
+    public NoLocalesInJavadocUrlDiagnosticRule() {
+        languageTags = Arrays.stream(Locale.getAvailableLocales())
```

Comment:
Would it be better to make this into a regex?

```java
StringBuilder patternBuilder = new StringBuilder(".*?\\/(");

Locale[] availableLocales = Locale.getAvailableLocales();
for (int i = 0; i < availableLocales.length; i++) {
    if (i > 0) {
        patternBuilder.append('|');
    }
    
    patternBuilder.append(availableLocale[i].toLanguageTag());
}

patternBuilder.append(")\\/.*");

localePattern = Pattern.compile(patternBuilder.toString(), Pattern.CASE_INSENSITIVE);

Reply:
regex is always better - I just pity the next person who has to maintain the code! :)

Reply:
In this case it is nice to iterate over each, as then we know which language tag is the issue and we can report that to the user to make search / replace easier.

Reply:
Agreed, realized after I wrote this that the current pattern is better as it allows multiple instances to be marked.